cg/store/crud/read.py

```diff
@@ -497,6 +498,25 @@ def get_illumina_sequencing_run_by_device_internal_id(
             device_internal_id=device_internal_id,
         ).first()
 
+    def get_latest_illumina_sequencing_run_for_case(
+        self, case_internal_id: str
+    ) -> IlluminaSequencingRun:
+        """Get Illumina sequencing run entry by case internal id."""
+        case: Case = self.get_case_by_internal_id(case_internal_id)
+        samples_on_case: list[Sample] = case.samples
+        sample_metrics: list[SampleRunMetrics] = []
+        for sample in samples_on_case:
+            sample_metrics.extend(sample.sample_run_metrics)
+        sequencing_runs: list[IlluminaSequencingRun] = [
+            apply_illumina_sequencing_run_filter(
+                runs=self._get_query(IlluminaSequencingRun),
+                filter_functions=[IlluminaSequencingRunFilter.BY_ENTRY_ID],
+                entry_id=sample_metric.instrument_run_id,
+            ).first()
+            for sample_metric in sample_metrics
+        ]
+        return max(sequencing_runs, key=lambda run: run.sequencing_completed_at)
```

Comment:
I think the logic of this function makes sense but the business logic of looking at the lates sequencing run of a case when starting is very strange. Basically this should only be used for NIPT. Can we make that clear in both the function name and doc string?

```suggestion
    def get_latest_illumina_sequencing_run_for_nipt_case(
        self, case_internal_id: str
    ) -> IlluminaSequencingRun:
        """Get Illumina sequencing run entry by case internal id. All samples in a NIPT case are run on the same flow cell hence there should only ever be one sequencing run."""
        case: Case = self.get_case_by_internal_id(case_internal_id)
        samples_on_case: list[Sample] = case.samples
        sample_metrics: list[SampleRunMetrics] = []
        for sample in samples_on_case:
            sample_metrics.extend(sample.sample_run_metrics)
        sequencing_runs: list[IlluminaSequencingRun] = [
            apply_illumina_sequencing_run_filter(
                runs=self._get_query(IlluminaSequencingRun),
                filter_functions=[IlluminaSequencingRunFilter.BY_ENTRY_ID],
                entry_id=sample_metric.instrument_run_id,
            ).first()
            for sample_metric in sample_metrics
        ]
        return max(sequencing_runs, key=lambda run: run.sequencing_completed_at)
```

Maybe it also makes sense to raise an error if there are more than one sequencing run for a nipt-case? That would signify that our assumption is false?

Reply:
This might be radical but should we raise and error if `case.data_analysis!=Workflow.FLUFFY`. I'm scare that this code will be used in cases where it is nonsensical